src/command/restore/restore.c

```diff
@@ -528,10 +531,16 @@ restoreManifestMap(Manifest *const manifest)
                     // Remap tablespace if a mapping was found
                     if (tablespacePath != NULL)
                     {
-                        LOG_INFO_FMT("map tablespace '%s' to '%s'", strZ(target->name), strZ(tablespacePath));
+                        // Append dbid to the path for Greenplum
+                        const String *const fixedTablespacePath =
+                            cfgOptionStrId(cfgOptFork) == CFGOPTVAL_FORK_GPDB ?
+                                strNewFmt("%s/%u", strZ(tablespacePath), manifestData(manifest)->pgId) : tablespacePath;
```

Comment:
Do we really need new variable?
```suggestion
                        if (cfgOptionStrId(cfgOptFork) == CFGOPTVAL_FORK_GPDB)
                                tablespacePath = strNewFmt("%s/%u", strZ(tablespacePath), manifestData(manifest)->pgId);
```